### PR TITLE
major update on cbo tutorials

### PR DESCRIPTION
**Pull request by Liebherr**
old version https://developers.sap.com/tutorials/abap-extensibility-cbo-create.html
new version https://developers-qa-blue.wcms-nonprod.c.eu-de-2.cloud.sap/tutorials/abap-extensibility-cbo-create.html

### CODE REVIEW

**Comment by Liebherr** [comment]
Hi Arianna, thanks for reviewing, hope its easily possible with the diff tools.
Thanks and Regards
Ulrike

**Comment by Liebherr** [comment]
list of old vs new tutorial versions

cbo creation
- old https://developers.sap.com/tutorials/abap-extensibility-cbo-create.html
- new https://developers-qa-blue.wcms-nonprod.c.eu-de-2.cloud.sap/tutorials/abap-extensibility-cbo-create.html

cbo ui generation
- old https://developers.sap.com/tutorials/abap-extensibility-cbo-ui-generation.html
- new https://developers-qa-blue.wcms-nonprod.c.eu-de-2.cloud.sap/tutorials/abap-extensibility-cbo-ui-generation.html

cbo logic
- old https://developers.sap.com/tutorials/abap-extensibility-cbo-logic.html
- new https://developers-qa-blue.wcms-nonprod.c.eu-de-2.cloud.sap/tutorials/abap-extensibility-cbo-logic.html

cbo ui adaptation
- old abap-extensibility-cbo-ui-adaptation.html
- new https://developers-qa-blue.wcms-nonprod.c.eu-de-2.cloud.sap/tutorials/abap-extensibility-cbo-ui-adaptation.html

cbo calls external service
- old https://developers.sap.com/tutorials/abap-extensibility-cbo-execute-outbound-service.html
- new https://developers-qa-blue.wcms-nonprod.c.eu-de-2.cloud.sap/tutorials/abap-extensibility-cbo-execute-outbound-service.html

File: tutorials/abap-extensibility-cbo-create/abap-extensibility-cbo-create.md [hunk 1]
```diff
@@ -2,19 +2,15 @@
...
 
 ## Prerequisites  
-<<<<<<< HEAD
- - **Authorizations:** Your user needs a business role with business catalog **Extensibility** (ID: `SAP_CORE_BC_EXT`) in your **S/4HANA Cloud** system
-=======
  - **Authorizations:** Your user needs a business role with business catalog **Extensibility – Custom Business Objects** (ID: `SAP_CORE_BC_EXT_CBO`) in your **S/4HANA Cloud** system
->>>>>>> 59f95048a11e62962d5c8eb49e89b6f027533a25
 
 ## You will learn
 - Define a custom business object and its a simple data structure
```
Comment (AriannaMussoBarcucci) [inline tutorials/abap-extensibility-cbo-create/abap-extensibility-cbo-create.md:16]: How to define a custom business object and its simple data structure

Comment (AriannaMussoBarcucci) [inline tutorials/abap-extensibility-cbo-create/abap-extensibility-cbo-create.md:16]: - How to define a custom business object and its (a?) simple data structure
- How to create a database table for the custom business object

File: tutorials/abap-extensibility-cbo-create/abap-extensibility-cbo-create.md [hunk 2]
```diff
@@ -38,9 +34,9 @@ In the first parts a manager wants to define business objects "Bonus Plan" for e
...
 1. Enter your SAP S/4HANA system's Fiori Launchpad.
 
-2. Open application **Custom Business Objects** from **Extensibility** Launchpad group.
+2. Start typing **Custom Business Objects** in the Launchpad search and open the App from the results.
 
-    ![Custom Business Objects application tile](tile_CBO.png)
+    ![Custom Business Objects application from search results](FLP_search_resultCBO.png)
 
 
 ### Create a custom business object
```
Comment (AriannaMussoBarcucci) [inline tutorials/abap-extensibility-cbo-create/abap-extensibility-cbo-create.md:42]: Press new to start the creation of a custom business object.

The other editable fields will be automatically filled up with suggested values, but can be changed.

File: tutorials/abap-extensibility-cbo-create/abap-extensibility-cbo-create.md [hunk 3]
```diff
@@ -104,6 +100,7 @@ In the first parts a manager wants to define business objects "Bonus Plan" for e
 **Publish** the business object `Bonus Plan`.
 
 ![Press Publish](CBO_pressPublish.png)
+
 This creates the database persistence of the business object.
```
Comment (AriannaMussoBarcucci) [inline tutorials/abap-extensibility-cbo-create/abap-extensibility-cbo-create.md:106]: In the 'Test yourself' section:

data base --> database

**Review by AriannaMussoBarcucci — COMMENTED** [review]
First review done for cbo creation, as soon as you fix the typos I will merge :)

File: tutorials/abap-extensibility-cbo-ui-generation/abap-extensibility-cbo-ui-generation.md [hunk 4]
```diff
@@ -25,89 +25,68 @@ At the end you will have a running application that you can create, update and d
...
 
 
-### Expose UI as Application
+### Grant Access to Application
 
 
-Now you make the UI available as SAP Fiori Launchpad application by assigning it to a Business Catalog which corresponds to a group in Launchpad.
-From the Business Object's overview go to Custom Catalog Extension application by clicking the **Maintain Catalogs** action.
+Now you grant access to the generated UI by assigning it to a Business Catalog.
+Ensure not to be in edit mode which is the case just after publishing.
```
Comment (AriannaMussoBarcucci) [inline tutorials/abap-extensibility-cbo-ui-generation/abap-extensibility-cbo-ui-generation.md:60]: Missing space between 'Catalog.Ensure'.
Missing space between 'publishing.From'.

Comment (AriannaMussoBarcucci) [inline tutorials/abap-extensibility-cbo-ui-generation/abap-extensibility-cbo-ui-generation.md:60]: Missing space between 'Catalog.Ensure'.
Missing space between 'publishing.From'.

**Review by AriannaMussoBarcucci — COMMENTED** [review]
Ok, second review (cbo ui generation) done

**Review by AriannaMussoBarcucci — COMMENTED** [review]
Third review (cbo logic) done

**Comment by AriannaMussoBarcucci** [comment]
Review of cbo ui adaptation done - no comments here

File: tutorials/abap-extensibility-cbo-execute-outbound-service/abap-extensibility-cbo-execute-outbound-service.md [hunk 5]
```diff
@@ -130,286 +128,267 @@ In order to allow communication with the SAP API Business Hub Sandbox you have t
...
 1. Switch to **Logic** section.
 
-    ![Switch to Logic section](CBO_LogicSection.png)
-
 2. Enter the **After Modification** Event Logic.
 
     ![Enter After Modification logic](CBO_go2AfterModify.png)
 
+3. Code HTTP client creation
+   In order to call an external service from there you need to create an HTTP client in your custom business object logic.
```
Comment (AriannaMussoBarcucci) [inline tutorials/abap-extensibility-cbo-execute-outbound-service/abap-extensibility-cbo-execute-outbound-service.md:249]: confusing sentence: is there supposed to be a full stop after 'Code HTTP client creation'?

File: tutorials/abap-extensibility-cbo-execute-outbound-service/abap-extensibility-cbo-execute-outbound-service.md [hunk 6]
```diff
@@ -130,286 +128,267 @@ In order to allow communication with the SAP API Business Hub Sandbox you have t
...
-    * Check if the outbound service is available
-    CHECK cl_ble_http_client=>is_service_available(
-        communication_scenario = 'YY1_MACHINE_TRANSLATION_API'
-        outbound_service       = 'YY1_OS_MACHINE_TRANSLATION_API_REST'
-    ) = abap_true.
-    ```
-3. Implement creation of HTTP client
+    - In the already existing coding go to the end but stay in front of the last ENDIF
+  
+    - Implement a check if the outbound service is available
```
Comment (AriannaMussoBarcucci) [inline tutorials/abap-extensibility-cbo-execute-outbound-service/abap-extensibility-cbo-execute-outbound-service.md:253]: re-phrasing suggestion: Implement the following code to check whether the outbound service is available

File: tutorials/abap-extensibility-cbo-execute-outbound-service/abap-extensibility-cbo-execute-outbound-service.md [hunk 7]
```diff
@@ -130,286 +128,267 @@ In order to allow communication with the SAP API Business Hub Sandbox you have t
...
+        ).
+        ```
 
-### Code request body string
+4. Code request body string
 
-Implement the creation of the Request Body.
-As the aim is to translate every other language than english into english, the target language is set to english. The source language and the to be translated feedback are gotten from the corresponding fields of the custom business object.
-The request body in JSON format looks this way.
+    Implement the creation of the Request Body.
```
Comment (AriannaMussoBarcucci) [inline tutorials/abap-extensibility-cbo-execute-outbound-service/abap-extensibility-cbo-execute-outbound-service.md:274]: Missing space between 'Request Body.As the aim'

File: tutorials/abap-extensibility-cbo-execute-outbound-service/abap-extensibility-cbo-execute-outbound-service.md [hunk 8]
```diff
@@ -130,286 +128,267 @@ In order to allow communication with the SAP API Business Hub Sandbox you have t
...
+    As the aim is to translate every other language than english into english, the target language is set to english. The source language and the to be translated feedback are gotten from the corresponding fields of the custom business object.
+    The request body in JSON format looks this way.
```
Comment (AriannaMussoBarcucci) [inline tutorials/abap-extensibility-cbo-execute-outbound-service/abap-extensibility-cbo-execute-outbound-service.md:276]: Missing space between 'business object.The request body'

File: tutorials/abap-extensibility-cbo-execute-outbound-service/abap-extensibility-cbo-execute-outbound-service.md [hunk 9]
```diff
@@ -130,286 +128,267 @@ In order to allow communication with the SAP API Business Hub Sandbox you have t
...
-    	WHEN OTHERS.
-    * Error handling
-    	ENDCASE.
-    ENDTRY.
-```
-5. Publish the After Modification logic.
+7. Publish the After Modification logic.
 
 
 ### Test the application
```
Comment (AriannaMussoBarcucci) [inline tutorials/abap-extensibility-cbo-execute-outbound-service/abap-extensibility-cbo-execute-outbound-service.md:387]: the 'Test Yourself' section is empty

**Review by AriannaMussoBarcucci — COMMENTED** [review]
Last review (cbo execute outbound' done :)

File: tutorials/abap-extensibility-cbo-ui-generation/abap-extensibility-cbo-ui-generation.md [hunk 10]
```diff
@@ -25,89 +25,68 @@ At the end you will have a running application that you can create, update and d
...
 
 Start adding a catalog with the **Add** action.
 
-<!-- border -->![Add new Custom Catalog Extension](CCE_add.png)
+![Add new Custom Catalog Extension](CCE_add.png)
 
 In the opening value help narrow down the result list by searching for `Custom Business`, select the Catalog with role ID `SAP_CORE_BC_EXT_CBO` and press **OK**.
 
-<!-- border -->![Value Help for adding Custom Catalog Extension](CCE_addValueHelp.png)
+![Value Help for adding Custom Catalog Extension](CCE_addValueHelp.png)
```
[the remark this answers is not in the document]
Comment (Liebherr) [reply tutorials/abap-extensibility-cbo-ui-generation/abap-extensibility-cbo-ui-generation.md:73]: the problem are not the black marks in the corners but the missing mandatory border around the rest ; )

File: tutorials/abap-extensibility-cbo-ui-generation/abap-extensibility-cbo-ui-generation.md [hunk 11]
```diff
@@ -25,89 +25,68 @@ At the end you will have a running application that you can create, update and d
...
-
 ### Test Bonus Plan application
 
 
-**Open** the Bonus Plan application.
+Start typing **Bonus Plans** in the Launchpad search and open the App from the results.
+
+![Bonus Plans application from search results](FLP_search_resultBonusPlans.png)
 
 **Create** an object.
```
[the remark this answers is not in the document]
Comment (Liebherr) [reply tutorials/abap-extensibility-cbo-ui-generation/abap-extensibility-cbo-ui-generation.md:91]: no idea why, need to check

File: tutorials/abap-extensibility-cbo-execute-outbound-service/abap-extensibility-cbo-execute-outbound-service.md [hunk 12]
```diff
@@ -59,15 +59,11 @@ To get to know the Machine Translation API service first, you can try it out in
...
+6. Exchange the default body with the code from before.
 
-6. Scroll down to the **Try Out** button and press it
-
-    ![Button to try out the service](GetServiceEndPoint_TryOutButton.png)
-
-    The response to the service call will appear below.
+7. Hit the **Run** button.
+   
+8. The **RESPONSE** to the service call will appear.
```
[the remark this answers is not in the document]
Comment (Liebherr) [reply tutorials/abap-extensibility-cbo-execute-outbound-service/abap-extensibility-cbo-execute-outbound-service.md:66]: actually the screenshot at step 1 shows this as well

File: tutorials/abap-extensibility-cbo-execute-outbound-service/abap-extensibility-cbo-execute-outbound-service.md (hunk 6 above)
Comment (Liebherr) [reply tutorials/abap-extensibility-cbo-execute-outbound-service/abap-extensibility-cbo-execute-outbound-service.md:253]: well, it shall somehow still be a task to try doing yourself not just copy and paste

File: tutorials/abap-extensibility-cbo-ui-generation/abap-extensibility-cbo-ui-generation.md [hunk 13]
```diff
@@ -25,89 +25,68 @@ At the end you will have a running application that you can create, update and d
...
-<!-- border -->![Add new Custom Catalog Extension](CCE_add.png)
+![Add new Custom Catalog Extension](CCE_add.png)
 
 In the opening value help narrow down the result list by searching for `Custom Business`, select the Catalog with role ID `SAP_CORE_BC_EXT_CBO` and press **OK**.
 
-<!-- border -->![Value Help for adding Custom Catalog Extension](CCE_addValueHelp.png)
+![Value Help for adding Custom Catalog Extension](CCE_addValueHelp.png)
 
->You could also choose another Catalog, but be aware that your user must have a Business Role containing the same to be able to view the created application.
+>You could also choose another Catalog, but be aware that your user must have a Business Role containing the same to be able to use the created application.
```
[the remark this answers is not in the document]
Comment (AriannaMussoBarcucci) [reply tutorials/abap-extensibility-cbo-ui-generation/abap-extensibility-cbo-ui-generation.md:75]: Perfect! Your re-phrasing is even clearer :)

File: tutorials/abap-extensibility-cbo-execute-outbound-service/abap-extensibility-cbo-execute-outbound-service.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (AriannaMussoBarcucci) [reply tutorials/abap-extensibility-cbo-execute-outbound-service/abap-extensibility-cbo-execute-outbound-service.md:66]: Oh good point!

File: tutorials/abap-extensibility-cbo-execute-outbound-service/abap-extensibility-cbo-execute-outbound-service.md (hunk 6 above)
Comment (AriannaMussoBarcucci) [reply tutorials/abap-extensibility-cbo-execute-outbound-service/abap-extensibility-cbo-execute-outbound-service.md:253]: Good point, then maybe you could re-phrase as: Implement the code to check whether the outbound service is available.
And then you could put the code in a hint box. Or write something like 'It should look something like this: CODE BLOCK HERE'

Comment (Liebherr) [reply tutorials/abap-extensibility-cbo-execute-outbound-service/abap-extensibility-cbo-execute-outbound-service.md:253]: but its like that at many more places. I will simply leave it like that.

**Comment by Liebherr** [comment]
Hi @AriannaMussoBarcucci , can you please do the final review and merge then : )
The Test Yourself issue was solved, but stays on QA only, see

Joshua: The reason we do not copy the rules.vr file to production is that the production repo is public and that would make the answer exposed.  The parser works to retrieve the validation questions from the Contribution repo to display them when redering in the production page.

**Pull request merged**